crates/solidity/outputs/cargo/tests/src/doc_examples/visitor_api.rs

```diff
@@ -4,10 +4,11 @@ use anyhow::{bail, ensure, Error, Result};
 use semver::Version;
 
 use slang_solidity::{
+    cst::{Node, RuleNode},
```

Comment:
I wonder why was the `syntax` namespace deleted? These are all syntax related APIs, and we originally intended to separate them from future APIs (semantics/codegen/etc..). For example, we might want to produce kinds or cursors that are not related to the syntax tree.

Reply:
remapped

Reply:
.. to match the rust APIs. 'syntax' doesn't sit right as a top level distinction.

Reply:
While I don't know what you discussed and intended earlier, I agree that `syntax` does seems like a good module boundary (at least for now, maybe AST will force us to rethink that?) with CST-specific cursor/visitor or grammar-specific types like ProductionKind/RuleKind rather than being exported at top-level here.

Reply:
It's not in the rust namespace hierarchy, so I made the TS match the Rust.

I'm not a fan of `syntax` because I think that's the wrong categorisation. There's not much beyond (`morphology` and) `syntax` and `semantics`.

I think `cst` is a good top level namespace.

Reply:
I admit that most of the others belong under cst. But I would rather think about that outside of the context of this PR.

Reply:
> I'm not a fan of syntax because I think that's the wrong categorisation. There's not much beyond (morphology and) syntax and semantics.

> I admit that most of the others belong under cst. But I would rather think about that outside of the context of this PR.

Happy to hear your thoughts on this in the next meeting, and perhaps consider a different namespace for the future, but I agree, as you said, let's keep namespace changes outside the context of this PR?

Reply:
Sure, but as it is, not doing extra work to get back to `syntax/` or `syntax/nodes/`.

My current thinking is that `fe` (front-end) and `be` (back-end) is a better distinction than `syntax` and `semantics`.

I would like to immediately put in the nanopass langage identifier. In the IU system they use e.g. `Lsrc` and `Lremove-redundant-loads`, as pass identifiers, refering to the language type / node type namespace. Each pass has a slightly different set of node types, many of which are simply re-exports of the previous pass.

The cst is a slightly different kettle of fish, but I think we can include the partial charateristic of the types as a transformation over the types derived from the grammar - which should be renamed to `language`, and the `Language` we currently have should be `Parser`, and it should be used in conjunction with a factory called `Solidity` (name derived from the definition) that creates these objects.

Reply:
Happy to merge this for now, assuming we take care of namespaces the first thing after. Since this blocks us from releasing bug fixes (existing and new ones).

Thinking of front-end/back-end distinction, I think most of our work in the foreseeable future can be categorized as front end, or analysis APIs (middleware). I mostly see backend used in the context of optimizations/codegen, which is not on our roadmap. How about making the division about user operations/features instead? like `scanning`, `parsing`, and `binding` (for scope graphs).

Let's brainstorm in the next stand up.